Data science assignment 4 Zain Ul Haq(zh05616@st.habib.edu.pk)

In [1]:
import pandas as pd 
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from scipy import stats

#importing all the libraries that we need to compile the code

data=pd.read_csv("EmployeeAttrition.csv")

#reading the csv


We want to compare how much money men and women make each month. We'll do this by taking samples from both groups and doing some tests.

The null hypothesis is that men and women make about the same amount of money each month.

Our alternative hypothesis is that men might make more or less than women.

In [2]:
male = data[(data['Gender']=='Male')]['MonthlyIncome']
female = data[(data['Gender']=='Female')]['MonthlyIncome']
stats.ttest_ind(male, female)

#this formula gives us the p value and other stats 

TtestResult(statistic=-1.2212617308870655, pvalue=0.22218303455087904, df=1468.0)

In summary, based on the obtained p-value of 0.22, which is greater than 0.05, we do not have sufficient evidence to reject the Null Hypothesis. 
This suggests that there is no statistically significant difference in the monthly income between males and females.

q2

Null Hypothesis: The work life balance of males is less than the work life balance of females


Alternative Hypothesis: The work life balance of males is not less than the work life balance of females


In [3]:
male = data[(data['Gender']=='Male')]['WorkLifeBalance']
female = data[(data['Gender']=='Female')]['WorkLifeBalance']
stats.ttest_ind(male, female)

TtestResult(statistic=-0.10546787956064164, pvalue=0.9160189649366257, df=1468.0)

In summary, as our p-value is 0.91 and therefore greater than 0.05, we do not have enough evidence to reject the Null Hypothesis. This implies that there is no statistically significant difference in work-life balance between males and females.

q3

Null Hypothesis: The years at company for single people is less than years at company for married people

Alternative Hypothesis: The years at company for single people is not less than years at company for married people

In [4]:
single = data[(data['MaritalStatus']=='Single')]['YearsAtCompany']
married = data[(data['MaritalStatus']=='Married')]['YearsAtCompany']
stats.ttest_ind(single, married)

TtestResult(statistic=-2.5589985737464476, pvalue=0.010625532709863163, df=1141.0)


In summary, based on the obtained p-value of 0.01, which is less than 0.05, we have enough evidence to reject the Null Hypothesis and, as a result, accept the Alternative Hypothesis. This indicates that the years at the company for single people are not less than the years at the company for married people.

q4

Null Hypothesis: If the environment satisfaction of the poeple who want to leave the company is less than the enviroment satisfaction of people who don't want to leave the company


Alternative Hypothesis: If the environment satisfaction of the poeple who want to leave the company is not less than the enviroment satisfaction of people who don't want to leave the company.

In [10]:
yes = data[(data['Attrition']=='Yes')]['EnvironmentSatisfaction']
no = data[(data['Attrition']=='No')]['EnvironmentSatisfaction']
stats.ttest_ind(yes, no)

q5

Null Hypothesis: If the montly income of the manager is greater than the monthly income of the laboratory technician


Alternative Hypothesis: If the montly income of the manager is not greater than the monthly income of the laboratory technician

In [5]:
manager = data[(data['JobRole']=='Manager')]['MonthlyIncome']
technician = data[(data['JobRole']=='Laboratory Technician')]['MonthlyIncome']
stats.ttest_ind(manager,technician)


TtestResult(statistic=76.03317414634023, pvalue=1.9942336668736867e-223, df=359.0)

In summary, based on the extremely small p-value we obtained (1.99^e-(223)), which is far less than 0.05, we have strong evidence to reject the Null Hypothesis and, instead, accept the Alternative Hypothesis. This indicates that the monthly income of the manager is not greater than the monthly income of the laboratory technician.

q6

Null Hypothesis: Gender and department are two variables that are not associated with each other


Alternative: Gender and department are associated with each other

In [14]:
chisqt = pd.crosstab(data['Department'],data['Gender'], margins=True)

In [15]:
value = np.array([chisqt.iloc[0][0:].values,
                  chisqt.iloc[1][0:].values])

In [16]:
stats.chi2_contingency(value)

Chi2ContingencyResult(statistic=1.4709421142834458, pvalue=0.47927963769548776, dof=2, expected_freq=array([[ 24.54785156,  38.45214844,  63.        ],
       [374.45214844, 586.54785156, 961.        ]]))

As evident from the provided values, the second value represents the p-value, which is 0.47. This result suggests that we should not reject the null hypothesis. Therefore, we can conclude that there is no significant association between Gender and Department; they are independent variables.